File: deal/src/main/java/ru/artemev/deal/dto/LoanApplicationRequestDTO.java
package ru.artemev.deal.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import javax.validation.constraints.DecimalMin;
import javax.validation.constraints.Email;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Past;
import javax.validation.constraints.Pattern;
import javax.validation.constraints.Size;
import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@Schema(description = "Заявка на кредит")
public class LoanApplicationRequestDTO {
  @NotNull
  @DecimalMin(value = "100000", message = "Сумма не может быть меньше 100 000р")
  @Schema(description = "Сумма кредита", example = "100000")
  private BigDecimal amount;

  @Min(value = 6, message = "Срок займа не может быть меньше 6 месяцев")
  @NotNull
  @Schema(description = "Срок займа", example = "12")
  private Integer term;

  @NotBlank
  @Size(min = 2, max = 100)
  @Schema(description = "Имя", example = "Петя")
  private String firstName;

  @NotBlank
  @Size(min = 2, max = 100)
  @Schema(description = "Фамилия", example = "Петров")
  private String lastName;

  @NotBlank
  @Size(min = 2, max = 100)
  @Schema(description = "Отчество", example = "Петрович")
  private String middleName;

  @Email
  @Schema(description = "Электронная почта", example = "ppetrov@example.com")
  private String email;

  @NotNull
  @Past(message = "Дата может быть только прошлым")
  private LocalDate birthday;

  @NotNull
  @Pattern(regexp = "[\\d]{4}", message = "Не верно введена серия паспорта")
  @Schema(description = "Серия паспорта", example = "1234")
  private String passportSeries;

  @NotNull
  @Pattern(regexp = "[\\d]{6}", message = "Не верно введен номер паспорта")
  @Schema(description = "Номер пасорта", example = "567890")
  private String passportNumber;
}

File: deal/src/main/kotlin/ru/kit/deal/dto/LoanApplicationRequestDTO.kt
package ru.kit.deal.dto

import io.swagger.v3.oas.annotations.media.Schema
import java.math.BigDecimal
import java.time.LocalDate
import javax.validation.constraints.*

@Schema(description = "Заявка на кредит")
data class LoanApplicationRequestDTO(
    @NotNull
    @DecimalMin(value = "100000", message = "Сумма не может быть меньше 100 000р")
    @Schema(description = "Сумма кредита", example = "100000")
    val amount: BigDecimal? = null,

    @Min(value = 6, message = "Срок займа не может быть меньше 6 месяцев")
    @NotNull
    @Schema(description = "Срок займа", example = "12")
    val term: Int? = null,

    @NotBlank
    @Size(min = 2, max = 100)
    @Schema(description = "Имя", example = "Петя")
    val firstName: String? = null,

    @NotBlank
    @Size(min = 2, max = 100)
    @Schema(description = "Фамилия", example = "Петров")
    val lastName: String? = null,

    @NotBlank
    @Size(min = 2, max = 100)
    @Schema(description = "Отчество", example = "Петрович")
    val middleName: String? = null,

    @Email
    @Schema(description = "Электронная почта", example = "ppetrov@example.com")
    val email: String? = null,

    @NotNull
    @Past(message = "Дата может быть только прошлым")
    val birthday: LocalDate? = null,

    @NotNull
    @Pattern(regexp = "[\\d]{4}", message = "Не верно введена серия паспорта")
    @Schema(description = "Серия паспорта", example = "1234")
    val passportSeries: String? = null,

    @NotNull
    @Pattern(regexp = "[\\d]{6}", message = "Не верно введен номер паспорта")
    @Schema(description = "Номер пасорта", example = "567890")
    val passportNumber: String? = null,
)